### DISCUSSION TITLE
metadata for dirty-since-last-submit

### DISCUSSION THREAD

**Discussion by santino** [react-hook-form/react-hook-form#4014] [Ideas] [answered]
Hi there,
I am evaluating migrating to React hook Form from Final Form.

One piece of metadata that would be useful for me is the dirty-since-last-submit.
My use-case is that I want to show submission errors (e.g. a generic error from the server) when submission fails.
Then I want to hide this error as soon as the user is updating any form value since that's most likely no longer applicable.

For example on a login form if the server returns "incorrect credentials" I want to show a generic error on the whole form and not a detailed error on each field (such as invalid username, or invalid password).
But as soon as the user understands that they made a mistake and so update any field to correct this I want to hide the error rather than keeping it there until they submit again.

Having metadata such as dirty-since-last-submit would be very useful in this use-case.

Alternatively, I would be happy to try and implement this on my own with a solution that makes sense.
The only thing I can think of right now is to watch for changes in all the fields and as soon as one value changes I can trigger `clearErrors`; but this doesn't seem a good option in terms of performance.

Thanks for your help!

**Comment by bluebill1049** [comment, marked as answer]
hey @santino would you like to send a feature request on this?

**Reply by santino** [reply]
Sure thing.
I wanted to check your feelings about something like this and see if you wanted to provide any suggestions to achieve this with current features.

Following your input, I have raised #4016

**Comment by bombillazo** [comment]
This is no possible natively with `rhf-plus`

https://github.com/bombillazo/rhf-plus/blob/master/docs/is-dirty-since-submit.md
